File: app/src/main/java/moises/com/usersapp/ui/base/BaseFragment.java
package moises.com.usersapp.ui.base;

import android.content.res.Configuration;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.appcompat.widget.Toolbar;

import moises.com.usersapp.R;

public class BaseFragment extends Fragment {
    private Toolbar toolbar;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);

        toolbar = (Toolbar)getActivity().findViewById(R.id.toolbar);
    }

    protected void setTitle(@NonNull String title){
        if(toolbar != null)
            toolbar.setTitle(title);
    }

    protected boolean isLandScape(){
        int currentOrientation = getResources().getConfiguration().orientation;
        return currentOrientation == Configuration.ORIENTATION_LANDSCAPE;
    }

    protected boolean isTablet(){
        return getActivity().getResources().getBoolean(R.bool.isTablet);
    }

    protected boolean isPhoneLarge(){
        return getActivity().getResources().getBoolean(R.bool.isPhoneLarge);
    }
}

File: app/src/main/java/moises/com/usersapp/ui/base/BaseFragment.kt
package moises.com.usersapp.ui.base

import android.content.res.Configuration
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

import moises.com.usersapp.R

open class BaseFragment : Fragment() {
    protected val isLandScape: Boolean
        get() {
            val currentOrientation = resources.configuration.orientation
            return currentOrientation == Configuration.ORIENTATION_LANDSCAPE
        }

    protected val isTablet: Boolean
        get() = requireActivity().resources.getBoolean(R.bool.isTablet)

    protected val isPhoneLarge: Boolean
        get() = requireActivity().resources.getBoolean(R.bool.isPhoneLarge)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    protected fun changeTitle(title: String) {
        val actionBar = (activity as AppCompatActivity).supportActionBar
        actionBar?.title = title
    }

    protected fun showMessageInToast(message: String) {
        Toast.makeText(context, message, Toast.LENGTH_LONG).show()
    }

    protected fun showError(throwable: Throwable) {
        Toast.makeText(context, throwable.localizedMessage, Toast.LENGTH_LONG).show()
    }
}
